File: app/src/main/java/com/ashlikun/loadswitch/simple/MainActivity.java
package com.ashlikun.loadswitch.simple;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;

import com.ashlikun.loadswitch.ContextData;
import com.ashlikun.loadswitch.LoadSwitch;
import com.ashlikun.loadswitch.LoadSwitchService;
import com.ashlikun.loadswitch.DefaultOnLoadLayoutListener;
import com.ashlikun.loadswitch.OnLoadSwitchClick;

public class MainActivity extends AppCompatActivity
        implements OnLoadSwitchClick {
    LoadSwitchService loadSwitchService;
    int currStatus = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        loadSwitchService = LoadSwitch
                .get()
                .register(findViewById(R.id.switchRoot), new DefaultOnLoadLayoutListener(this, this));
        findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (currStatus == 0) {
                    loadSwitchService.showEmpty(new ContextData().setImgSize(500, 500));
                    currStatus = 1;
                } else if (currStatus == 1) {
                    loadSwitchService.showLoading(new ContextData());
                    currStatus = 2;
                } else if (currStatus == 2) {
                    loadSwitchService.showContent();
                    currStatus = 3;
                } else if (currStatus == 3) {
                    loadSwitchService.showRetry(new ContextData());
                    currStatus = 0;
                }
            }
        });
    }

    @Override
    public void onRetryClick(ContextData data) {

    }

    @Override
    public void onEmptyClick(ContextData data) {

    }
}

File: app/src/main/java/com/ashlikun/loadswitch/simple/MainActivity.kt
package com.ashlikun.loadswitch.simple

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.ashlikun.loadswitch.*

class MainActivity : AppCompatActivity(), OnLoadSwitchClick {
    var loadSwitchService: LoadSwitchService? = null
    var currStatus = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        LoadSwitch.init(application)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        loadSwitchService = LoadSwitch().create(findViewById(R.id.twoRecyclerView), DefaultOnLoadLayoutListener(this, this))
        findViewById<View>(R.id.button).setOnClickListener { v: View? ->
            if (currStatus == 0) {
                loadSwitchService!!.showEmpty(ContextData(imgHeight = 500, imgWidth = 500))
                currStatus = 1
            } else if (currStatus == 1) {
                loadSwitchService!!.showLoading(ContextData())
                currStatus = 2
            } else if (currStatus == 2) {
                loadSwitchService!!.showContent()
                currStatus = 3
            } else if (currStatus == 3) {
                loadSwitchService!!.showRetry(ContextData())
                currStatus = 0
            }
        }
        findViewById<View>(R.id.button2).setOnClickListener { startActivity(Intent(this@MainActivity, Main2Activity::class.java)) }
    }

    override fun onRetryClick(data: ContextData) {}
    override fun onEmptyClick(data: ContextData) {}
}